zk-token-sdk/src/zk_token_elgamal/pod/auth_encryption.rs

```diff
@@ -1,16 +1,26 @@
 //! Plain Old Data types for the AES128-GCM-SIV authenticated encryption scheme.
 
-#[cfg(not(target_os = "solana"))]
-use crate::encryption::auth_encryption::{self as decoded, AuthenticatedEncryptionError};
 use {
     crate::zk_token_elgamal::pod::{Pod, Zeroable},
     base64::{prelude::BASE64_STANDARD, Engine},
     std::fmt,
 };
+#[cfg(not(target_os = "solana"))]
+use {
+    crate::{
+        encryption::auth_encryption::{self as decoded, AuthenticatedEncryptionError},
+        zk_token_elgamal::pod::ParseError,
+    },
+    std::str::FromStr,
+};
 
 /// Byte length of an authenticated encryption ciphertext
 const AE_CIPHERTEXT_LEN: usize = 36;
 
+/// Maximum length of a base64 encoded authenticated encryption ciphertext
+#[cfg(not(target_os = "solana"))]
```

Comment:
Is there a reason to not allow this on-chain? It's certainly not efficient and discouraged, but we don't prevent people from parsing strings as `Pubkey`s on-chain